ophyd/v2/epics.py

```diff
@@ -2,304 +2,113 @@
-        for value_listener in self._value_listeners:
-            value_listener(self._value)
-        for reading_listener in self._reading_listeners:
-            reading_listener({self.name: self._reading})
-
-    def _monitor_if_needed(self) -> None:
-        should_monitor = (
-            self._value_listeners or self._reading_listeners or self._staged
-        )
-        if should_monitor and not self._monitor:
-            # Start a monitor
-            self._monitor = self.read_channel.monitor_reading_value(self._callback)
-        elif self._monitor and not should_monitor:
-            # Stop the monitor
-            self._monitor.close()
-            self._valid.clear()
-            self._monitor = None
-            self._reading = None
-            self._value = None
-
-    def subscribe_value(self, function: Callback[T]) -> None:
-        self._value_listeners.append(function)
-        if self._value is not None:
-            function(self._value)
-        self._monitor_if_needed()
-
-    def subscribe(self, function: Callback[Dict[str, Reading]]) -> None:
-        self._reading_listeners.append(function)
-        if self._reading is not None:
-            function({self.name: self._reading})
-        self._monitor_if_needed()
-
-    def clear_sub(self, function: Callback):
-        try:
-            self._value_listeners.remove(function)
-        except ValueError:
-            self._reading_listeners.remove(function)
-        self._monitor_if_needed()
-
-    def stage(self) -> List[Any]:
-        self._staged = True
-        self._monitor_if_needed()
-        return [self]
-
-    def unstage(self) -> List[Any]:
-        self._staged = False
-        self._monitor_if_needed()
-        return [self]
-
-
-class EpicsSignalW(SignalW[T], _WithDatatype[T]):
-    """Writeable EPICS Signal backed by a single PV"""
-
-    def __init__(self, datatype: Type[T], write_pv: Optional[str] = None) -> None:
-        #: Request that underlying PV connection is made using this Python datatype
-        self.datatype = datatype
-        #: Write PV. Can be prefixed by passing prefix in `connect()`
-        self.write_pv = write_pv
-        #: Write `Channel`. A connected instance will be set in `connect()`
-        self.write_channel: Channel[T] = DISCONNECTED_CHANNEL
-
-    @property
-    def source(self) -> str:
-        return self.write_channel.source
-
-    async def connect(self, prefix: str = "", sim=False) -> None:
-        assert self.write_pv is not None, "Write PV not set"
-        self.write_channel = _make_channel_class(
-            prefix + self.write_pv, self.datatype, sim, self.write_channel
-        )
-        await self.write_channel.connect()
-
-    def set(self, value: T, wait=True) -> AsyncStatus:
-        return AsyncStatus(self.write_channel.put(value, wait=wait))
+    from ._aioca import CaSignalBackend
+except ImportError as ca_error:
 
+    def CaSignalBackend(*args, ca_error=ca_error, **kwargs):  # type: ignore
```

Comment:
This is valid, but it's more lines of code, and I think you would still need a bunch of `type: ignore` functions. The other option is to make a `RaisingSignalBackend` class that could be used instead of `CaSignalBackend`. I'm ambivalent about the best way to do this, I just went with the shortest lines of code that took me the least amount of time to fix. I'd be happy to accept a PR that did this in a nicer way after this one is merged